src/lib.rs

```diff
@@ -123,9 +126,7 @@ impl<V> IntMap<V> {
 
         vals.push((key, value));
 
-        if (self.count & 4) == 4 {
-            self.ensure_load_rate();
-        }
```

Comment:
What's the reason for checking `(self.count & 4) == 4`? Is it significantly more performant than the condition inside of `ensure_load_rate`?

Reply:
Tried to look at git log and just found a helpful "First version..." 8 years ago (yikes!) a good reminder to write better commit messages.
I remember having a conversion to a float before which is more expensive. An `and` operation should be a alot faster than one multiplication, division and comparison. But the extra branch might not be worth it.